x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/plugin/TransportEsqlQueryAction.java

```diff
@@ -190,8 +195,50 @@ private void innerExecute(Task task, EsqlQueryRequest request, ActionListener<Es
             executionInfo,
             remoteClusterService,
             planRunner,
-            listener.map(result -> toResponse(task, request, configuration, result))
+            ActionListener.wrap(result -> {
+                recordCCSTelemetry(task, executionInfo, request, null);
+                listener.onResponse(toResponse(task, request, configuration, result));
+            }, ex -> {
+                recordCCSTelemetry(task, executionInfo, request, ex);
+                listener.onFailure(ex);
+            })
         );
+
+    }
+
+    private void recordCCSTelemetry(Task task, EsqlExecutionInfo executionInfo, EsqlQueryRequest request, @Nullable Exception exception) {
+        if (executionInfo.isCrossClusterSearch() == false) {
+            return;
+        }
+
+        var usageBuilder = new CCSUsage.Builder();
+        usageBuilder.setClientFromTask(task);
+        if (exception != null) {
+            usageBuilder.setFailure(exception);
+        }
+        var took = executionInfo.overallTook();
+        if (took != null) {
+            usageBuilder.took(took.getMillis());
+        }
+        if (request.async()) {
+            usageBuilder.setFeature(CCSUsageTelemetry.ASYNC_FEATURE);
+        }
+        usageBuilder.setFeature(CCSUsageTelemetry.MRT_FEATURE);
```

Comment:
Not sure we need to do this really. Technically it's true but not interesting.

Reply:
It's a bit tricky because counting of mrt-is-true/false stats depends on it, and displaying the stats under "mrt=false" would be weird, but I think I'll drop those sections for ES|QL completely, since there's no MRT switch on it anyway.